### ISSUE TITLE
The stream cannot end when the blob has more than two parts

### ISSUE THREAD

**Issue by yarlajs** [denoland/deno#21507]
Version: Deno 1.38.5

```js
// test.js
import http from "node:http";
import stream from "node:stream";

const part1 = new Blob(["part1"]);
const part2 = new Blob(["part2"]);
const part3 = new Blob(["part3"]);

function request(blob) {
    const request = http.request({
        port: 3000,
        host: "localhost",
        path: "/",
        method: "POST"
    }, async function (res) {
        const m = stream.Readable.toWeb(res);
        const n = new Response(m);
        const t = await n.text();
        console.log(t);
    });
    request.useChunkedEncodingByDefault = true;
    const nodeStream = stream.Readable.from(blob.stream());
    nodeStream.pipe(request);
}

// request(part1);                          // work
// request(new Blob([part1, part2]));       // work
request(new Blob([part1, part2, part3]));   // not work (The node stream will not emit end event and the program will not exit)
// Even more parts cannot work

// `node test.js` work
// `deno run -A test.js` not work
```

**Comment by yarlajs** [comment]
> Thanks for the clear repro. Do you know if the same thing happens w/the fetch API? Regardless, definitely looks like a bug!

Fetch API can work (Because there is no need to convert web stream to node stream).

**Comment by kt3k** [comment]
This seems happened until 1.38.5 and fixed in 1.39.0.

I've bisected the main branch, and this commit ( https://github.com/denoland/deno/commit/e6e708e46c51f3154a81ed99cd35c3d5569930f9 ) seems to have fixed the issue

**Comment by bartlomieju** [comment]
Sounds good, I'm gonna close the issue for now. Thanks for checking @kt3k
